packages/next/src/utils.ts

```diff
@@ -432,14 +436,21 @@ export async function getDynamicRoutes({
             dest: route.dest?.replace(/($|\?)/, '.rsc$1'),
           });
 
-          routes.push({
-            ...route,
-            src: route.src.replace(
-              new RegExp(escapeStringRegexp('(?:/)?$')),
-              '(?:\\.action)(?:/)?$'
-            ),
-            dest: route.dest?.replace(/($|\?)/, '.action$1'),
-          });
+          // push .action rewrites for dynamic routes if support is available
+          // and the dynamic route is in the prerender manifest
+          if (
+            canUseActionRewrite &&
+            prerenderManifest.blockingFallbackRoutes[page]
```

Comment:
cc @ijjk is this a good way to scope down dynamic rewrites? Validated that the existing dynamic force-static test passed but I'm unfamiliar with what all this `blockingFallbackRoutes` property is used for.